index.js

```diff
@@ -97,7 +97,12 @@ function fastifyNext (fastify, options, next) {
       reply.raw.setHeader(headerName, headerValue)
     }
 
-    await app.render(request.raw, reply.raw, path, request.query)
+    if (typeof args === undefined) {
+      await app.render(request.raw, reply.raw, path, request.query)
+    } else {
+      delete request.query
```

Comment:
this should not be deleted

Reply:
I will remove the delete operation. I tried the below test for evaluating this commit, but it fails. The additional params are not available in the `res` object itself.

```
test('should decorate with next render function passing additional args/query params', async t => {
  // t.plan(2)

  const fastify = Fastify()
  t.teardown(() => fastify.close())

  await fastify.register(require('./index'))

  fastify.addHook('onRequest', (req, reply, done) => {
    reply.header('test-header', 'hello')
    done()
  })

  fastify.get('/hello', (req, reply) => {
    return reply.nextRender('/hello', { msg: 'hi' })
  })

  const res = await fastify.inject({
    url: '/hello',
    method: 'GET'
  })

  t.equal(res.statusCode, 200)
  t.equal(res.headers['test-header'], 'hello')
  t.equal(res.raw.req.url, '/hello?msg=hi')
})
```

Reply:
Look in https://github.com/fastify/light-my-request for which properties are available.